### PR TITLE
version: use Go 1.18's git stamping as default implementation

### LINKED ISSUE #81: version: come up with good default implementation
When we do tailscale binary releases, we stamp version info into the binaries with the Go linker.

But for people just using "go get" the default way, we don't have that info.

Since Go modules, we at least see the summary of all our deps (and Go's version itself; I'm running a devel build) automatically stamped into the binaries:

```
$ go version -m ~/bin/tailscaled
/home/bradfitz/bin/tailscaled: devel +e7f9e17b79 Tue Jan 28 22:08:43 2020 +0000
        path    tailscale.com/cmd/tailscaled
        mod     tailscale.com   (devel)
        dep     github.com/apenwarr/fixconsole  v0.0.0-20191012055117-5a9f6489cc29      h1:muXWUcay7DDy1/hEQWrYlBy+g0EuwT70sBHg65SeUc4=            
        dep     github.com/golang/groupcache    v0.0.0-20200121045136-8c9f03a8e57e      h1:1r7pUrabqp18hOBcwBwiTsbnFeTZHV9eER/QT5JVZxY=    
        dep     github.com/google/go-cmp        v0.4.0  h1:xsAVV57WRhGj6kEIi8ReJzQlHHqcBYCElAvkovg3B/4=
        dep     github.com/klauspost/compress   v1.9.8  h1:VMAMUUOh+gaxKTMk+zqbjsSjsIcUcL/LF4o63i82QyA=
        dep     github.com/mdlayher/netlink     v1.1.0  h1:mpdLgm+brq10nI9zM1BpX1kpDbh3NLl3RSnVq6ZSkfg=
        dep     github.com/pborman/getopt       v0.0.0-20190409184431-ee0cd42419d3      h1:YtFkrqsMEj7YqpIhRteVxJxCeC3jJBieuLr0d4C4rSA=
        dep     github.com/tailscale/wireguard-go       v0.0.0-20200213180345-a7c4b7719b1d      h1:LVJovgZxbmPxtY6kJm4vwMtk0HpcNeI+vU2jB3T8M40=
        dep     golang.org/x/crypto     v0.0.0-20200210222208-86ce3cb69678      h1:wCWoJcFExDgyYx2m2hpHgwz8W3+FPdfldvIgzqDIhyg=
        dep     golang.org/x/net        v0.0.0-20200202094626-16171245cfb2      h1:CCH4IOTTfewWjGOlSp+zGcjutRKlBEZQ6wTn8ozI/nI=
        dep     golang.org/x/oauth2     v0.0.0-20200107190931-bf48bf16ab8d      h1:TzXSXBo42m9gQenoE3b9BGiEpg5IG2JkU5FkPIawgtw=
        dep     golang.org/x/sys        v0.0.0-20200217220822-9197077df867      h1:JoRuNIf+rpHl+VhScRQQvzbHed86tKkqwPMV34T8myw=
        dep     rsc.io/goversion        v1.2.0  h1:SPn+NLTiAG7w30IRK/DKp1BjvpWabYgxlLp/+kx5J8w=
```

We can get that at runtime (using https://godoc.org/rsc.io/goversion/version) but for better or worse, that doesn't include that the Git hash of the top-level module, only the deps.

So one thing we could do is add a fake dep to another module (like github.com/tailscale/version-horizon) that we have a bot auto-advance every $INTERVAL, so at least we have a rough date range of what binary people are running when they're running unofficial binaries. Or, if we're already having a bot do this, avoid that indirection and just have the bot auto-commit the latest date to version/version.go every $INTERVAL instead.

Low priority, but we'll probably want better data in the future.

/cc @apenwarr @crawshaw @danderson

### PR DESCRIPTION
**Pull request by bradfitz**
No more manual version bumps!

Fixes #81

### CODE REVIEW

**Comment by bradfitz** [comment]
```
bradfitz@tsdev:~/src/tailscale.com$ go run ./cmd/tailscale debug hostinfo
{
  "IPNVersion": "1.23.0-dev20220316-t8b36a39fb-dirty",
  "OS": "linux",
  "OSVersion": "Debian 11.2 (bullseye); kernel=5.10.0-10-amd64",
  "Hostname": "tsdev",
  "GoArch": "amd64"
}

bradfitz@tsdev:~/src/tailscale.com$ go run ./cmd/tailscale version
1.23.0-dev20220316
  tailscale commit: 8b36a39fb5b2cdd6908a910ee818175035c830ca-dirty
  go version: go1.18
```

**Comment by bradfitz** [comment]
cc @josharian

File: version/version.go [hunk 1]
```diff
@@ -6,28 +6,64 @@
...
 var Short = ""
 
 func init() {
-	// If it hasn't been link-stamped with -X (via build_dist.sh or similar),
-	// then use the VERSION.txt file in the root and the date in the Long
-	// variable above which we occasionally bump by hand.
-	if Short == "" {
-		Long = strings.TrimSpace(tailscaleroot.Version) + "-" + Long
+	if Long != "" && Short != "" {
+		// Built in the recommended way, using build_dish.sh.
```
Comment (josharian) [inline version/version.go:32]: s/dish/dist/

**Review by josharian — APPROVED** [review]

File: version/version.go (hunk 1 above)
Comment (bradfitz) [reply version/version.go:32]: cook_dish.sh

install with `yum install`

**Pull request merged**